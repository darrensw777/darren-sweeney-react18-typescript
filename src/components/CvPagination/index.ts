/* istanbul ignore file */
export { default } from './CvPagination';
