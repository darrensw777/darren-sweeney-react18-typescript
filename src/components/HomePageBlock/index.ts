/* istanbul ignore file */
export { default } from './HomePageBlock';
