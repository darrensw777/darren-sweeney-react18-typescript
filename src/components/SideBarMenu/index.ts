/* istanbul ignore file */
export { default } from './SideBarMenu';
