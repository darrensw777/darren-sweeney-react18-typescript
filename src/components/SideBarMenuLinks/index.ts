/* istanbul ignore file */
export { default } from './SideBarMenuLinks';
