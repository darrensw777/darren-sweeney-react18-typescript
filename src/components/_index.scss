@import './SideBarMenu/SideBarMenu.scss';
@import './SocialShare/SocialShare.scss';
@import './HomePageBlock/home-page-block.scss';
@import './SideNavBlock/side-nav-block.scss';
@import './SideBarMenuLinks/side-bar-menu-links.scss'
