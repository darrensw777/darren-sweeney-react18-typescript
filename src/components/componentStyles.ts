import styled from 'styled-components/macro';
import { FONT_SIZE_XL } from 'utils/styling/typography';
import {
    MarginBottom,
    HeaderFont,
    BorderRadius,
    PaddingRight,
    MarginRight,
    BodyCopy,
    MarginTop,
    Padding,
} from 'utils/styling/mixins';
import { MARGIN, MENU_TRANSITION_SPEED } from 'utils/styling/variables';
import { BORDER_GREY, PRIMARY_COLOR, PRIMARY_GREY, SECONDARY_GREY, WHITE } from 'utils/styling/colors';

export const CvPaginationWrapper = styled.div`
    .pagination {
        display: flex;
        flex-wrap: wrap;
        ${MarginBottom};

        p,
        .buttons {
            display: flex;
            flex-basis: 100%;
            justify-content: center;

            button {
                background: transparent;
                ${HeaderFont};
                border: none;
                cursor: pointer;
                font-size: ${FONT_SIZE_XL};
            }
        }
    }
`;

export const HomePageBlockWrapper = styled.div`
    width: 49%;
    @media screen and (max-width: 560px) {
        width: 100%;
    }
    ${MarginBottom}
    ${BorderRadius}
    .homepage-block {
        display: inline-block;
        color: #fff;
        text-align: center;
        position: relative;
        z-index: 2;
        width: 100%;
        transition: all 0.3s ease;

        &:hover {
            -webkit-filter: brightness(0.8);
            filter: brightness(0.8);
        }

        .link {
            cursor: pointer;
            z-index: 2;
            width: 100%;

            .card-image {
                width: 200px;
                height: 200px;
                margin: 0 auto;
                border-radius: 50%;
                overflow: hidden;
                margin-top: 8%;
                margin-bottom: 8%;
                border: 6px solid #fff;

                img {
                    clip-path: circle();
                    width: 100%;
                }
            }

            .link-holder {
                margin-bottom: 8%;
                display: flex;
                align-items: center;
                justify-content: center;
                ${HeaderFont}

                svg {
                    margin-right: 10px;
                }
            }
        }

        a {
            @media screen and (max-width: 768px) {
                font-size: 24px;
            }

            color: #fff;
            font-size: 30px;
            cursor: pointer;
            text-decoration: none;
        }
    }
`;

export const SideBarMenuWrapper = styled.div`
    .sidebar-container {
        .sidebar-menu {
            background: ${WHITE};
            border-radius: 0 0 10px 10px;
            transition: all ${MENU_TRANSITION_SPEED} ease;
            position: absolute;
            top: -1350px;
            left: 0;
            z-index: 200;
            width: 100%;
            max-width: 768px;
            &.open {
                top: 0px;
            }

            border: 1px solid ${BORDER_GREY};
            text-align: center;

            .nav-block-container {
                ${BorderRadius}
                display: flex;
                overflow: hidden;
                flex-wrap: wrap;
                flex-direction: row;
                justify-content: space-evenly;
                ${MarginTop}
            }

            .name-logo {
                text-decoration: none;
                ${HeaderFont}
                font-size: 36px;
                color: ${PRIMARY_GREY};
                ${MarginBottom}
                display: block;
                margin-top: 32px;

                @media (max-width: 526px) {
                    text-align: right;
                    ${PaddingRight}
                    font-size: 30px;
                }
            }

            .close-menu-icon {
                position: absolute;
                top: 20px;
                right: 20px;
            }

            img {
                margin-bottom: 20px;
                width: 80%;
                margin-left: 40px;
            }

            ul.sub-links {
                outline: none;
                list-style: none;
                margin-bottom: 0;
                padding-left: 20px;

                & > li {
                    ${MarginRight}

                    a {
                        ${BodyCopy}
                        padding-left: 16px;
                        display: flex;
                        align-items: center;
                        border-left: 2px solid ${BORDER_GREY};
                        color: ${SECONDARY_GREY};
                        height: 40px;

                        &[aria-current='true'],
                        &:hover {
                            border-left: 2px solid ${PRIMARY_COLOR};
                            text-decoration: none;
                            color: ${PRIMARY_COLOR};
                        }
                    }
                }
            }
        }
    }
`;

export const SideNavBlockWrapper = styled.div`
    flex: 0 1 calc(50% - ${MARGIN});
    .side-nav-block {
        margin: 0 0 ${MARGIN} 0;
        align-items: center;

        a {
            text-decoration: none;

            .link,
            .icon {
                display: flex;
                ${Padding}
                ${HeaderFont}
                    justify-content: center;
                font-size: 22px;
            }

            .link {
                color: ${WHITE};
                font-size: 22px;
            }

            .icon {
                color: ${PRIMARY_GREY};
                font-size: 42px;
            }
        }
    }
`;
