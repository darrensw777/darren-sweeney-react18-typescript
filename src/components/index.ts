/* istanbul ignore file */
/*
 * Exporting all the components from the components folder,
 * components can be imported as the example below:
 * import { Logo, Avatar } from 'components';
 */

export { default as SideBarMenu } from './SideBarMenu';
//export { default as SocialShare } from './SocialShare';
export { default as HomePageBlock } from './HomePageBlock';
export { default as SideNavBlock } from './SideNavBlock';
export { default as SideBarMenuLinks } from './SideBarMenuLinks';
