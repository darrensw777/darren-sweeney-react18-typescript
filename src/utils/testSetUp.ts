import { render } from '@testing-library/react';
import { ReactElement } from 'react';
import { MemoryRouter } from 'react-router-dom';

const testSetUp = ({ Component }: { Component: ReactElement }) => {
    const { container, getByText, getByRole, asFragment, getAllByRole, getByTestId, getByTitle } = render(Component, {
        wrapper: MemoryRouter,
    });
    return {
        container,
        getByText,
        getByRole,
        asFragment,
        getAllByRole,
        getByTestId,
        getByTitle,
    };
};

export default testSetUp;
