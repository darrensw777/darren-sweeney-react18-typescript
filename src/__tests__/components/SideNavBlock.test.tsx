import { SideNavBlock } from 'components';
import testSetUp from 'utils/testSetUp';

describe('SideNavBlock component', () => {
    const sideNavBlockProps = {
        link: '/',
        title: 'Home',
        iconClass: 'fa fa-home',
        backgroundColor: '#2C1320',
        onClick: () => {},
    };

    let asFragment: any, container: any;
    beforeEach(() => {
        ({ asFragment, container } = testSetUp({
            Component: <SideNavBlock {...sideNavBlockProps} />,
        }));
    });

    test('matches snapshot', () => {
        expect(asFragment()).toMatchSnapshot();
    });

    test('renders the block', () => {
        const blocks = container.getElementsByClassName('side-nav-block');
        expect(blocks.length).toBe(1);
    });
});
