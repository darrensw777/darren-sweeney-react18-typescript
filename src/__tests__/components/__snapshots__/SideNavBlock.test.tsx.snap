// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SideNavBlock component matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="side-nav-block"
  >
    <a
      aria-label="Home"
      class="link"
      href="/"
      role="navigation"
    >
      <div
        class="icon"
      >
        <i
          aria-hidden="true"
          class="fa fa-home"
        />
      </div>
      <div
        class="link"
      >
        Home
      </div>
    </a>
  </div>
</DocumentFragment>
`;
